### PR TITLE
Add show() and hide() methods to Canvas Layer.

### LINKED ISSUE #58122: Godot 4: CanvasLayer missing show() & hide()
### Godot version

4.0.alpha2

### System information

Ubuntu

### Issue description

CanvasLayer now has a `visible` property like CanvasItem does.
like with CanvasItem, this should go along with `show()` and `hide()` methods.

### Steps to reproduce

n/a

### Minimal reproduction project

_No response_

### PR DESCRIPTION
**Pull request by Josephblt**
Added show() and hide() methods to Canvas Layer.
Updated Canvas Layer Doc API. 

Fixes #58122

### CODE REVIEW

File: scene/main/canvas_layer.cpp [hunk 1]
```diff
@@ -67,6 +67,14 @@ void CanvasLayer::set_visible(bool p_visible) {
 	}
 }
 
+void CanvasLayer::show() {
+    set_visible(true);
```
[the remark this answers is not in the document]
Comment (Josephblt) [reply scene/main/canvas_layer.cpp]: Sorry, my editor was set to spaces. Fix sent. Thanks.

File: scene/main/canvas_layer.cpp [hunk 2]
```diff
@@ -67,6 +67,14 @@ void CanvasLayer::set_visible(bool p_visible) {
 	}
 }
 
+void CanvasLayer::show() {
+	set_visible(true);
+}
+
+void CanvasLayer::hide() {
+	set_visible(false);
+}
```
Comment (akien-mga) [inline scene/main/canvas_layer.cpp:76]: You need to also bind the methods in `_bind_methods`, otherwise they're not part of the scripting API.

**Comment by akien-mga** [comment]
After adding the bindings, please rebase to squash the commits. See [PR workflow](https://docs.godotengine.org/en/latest/community/contributing/pr_workflow.html#the-interactive-rebase).

File: scene/main/canvas_layer.cpp (hunk 2 above)
Comment (Josephblt) [reply scene/main/canvas_layer.cpp:76]: Sorry. I got lazy and did not check other places before creating the PR. Fix sent.

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Comment by Josephblt** [comment]
Hello @akien-mga. Should I cherry-pick this to branch [3.x](https://github.com/godotengine/godot/tree/3.x)?

**Comment by akien-mga** [comment]
No need, it should cherry-pick cleanly in my next round of cherry-picks.

**Comment by akien-mga** [comment]
Cherry-picked for 3.5.

**Pull request merged**